plugins/http/http_listen_input.go

```diff
@@ -71,43 +72,56 @@ func defaultStarter(hli *HttpListenInput) (err error) {
 	return nil
 }
 
+func (hli *HttpListenInput) makeField(name string, value string) (field *message.Field, err error) {
+	field, err = message.NewField(name, value, "")
+	if err != nil {
+		hli.ir.LogError(fmt.Errorf("can't add field %s: %s", name, err))
+	}
+	return
+}
+
 func (hli *HttpListenInput) makePackDecorator(req *http.Request) func(*PipelinePack) {
 	packDecorator := func(pack *PipelinePack) {
 		pack.Message.SetType("heka.httpdata.request")
 		pack.Message.SetPid(hli.hekaPid)
 		pack.Message.SetSeverity(int32(6))
-		if field, err := message.NewField("Protocol", req.Proto, ""); err == nil {
+
+		// Host on which heka is running.
+		pack.Message.SetHostname(hli.hostname)
+		pack.Message.SetEnvVersion("1")
```

Comment:
I'm not sure if it makes sense to arbitrarily set this to "1", but it seemed preferable to leaving it blank.